actors/evm/src/interpreter/system.rs

```diff
@@ -182,21 +182,31 @@ impl<'r, BS: Blockstore, RT: Runtime<BS>> System<'r, BS, RT> {
     }
 
     /// Load the bytecode.
-    pub fn load_bytecode(&self) -> Result<Option<Bytecode>, ActorError> {
-        match &self.bytecode {
-            Some(cid) => {
-                let bytecode = self
-                    .rt
-                    .store()
-                    .get(cid)
-                    .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to read state")?
-                    .expect("bytecode not in state tree");
-                if bytecode.is_empty() {
-                    return Ok(None);
+    pub fn load_bytecode(&self, delegate: Option<Cid>) -> Result<Option<Bytecode>, ActorError> {
```

Comment:
I'd definitely call the parameter something like `override_bytecode` if we keep it. I'd kind of prefer to just have a helper function for loading bytecode, and make _this_ function always load the "real" bytecode, but I don't care too much.

I.e.

```
pub fn load_bytecode(rt: &RT, cid: Cid) -> Result<Bytecode, ActorError> { ... }

impl System {
    pub fn load_bytecode(&self) -> Result<Option<Bytecod>, ActorError> { self.bytecode.map(|k|load_bytecode(self.rt, k)).transpose()? }   
}
```

Reply:
uhm ok, although we'd lose the error message discriminant.

Reply:
We shouldn't? Basically, I'm saying:

1. Put the "if delegated" statement on the outside.
2. Extract the body of `load_bytecode` into a helper.
3. If delegated, call the helper. Otherwise, call `load_bytecode`.

Or am I missing something?

(but really, this _isn't_ critical, it's just me nitpicking)

Reply:
ok

Reply:
you have to deal with shit like this:
```
   Compiling fil_actor_evm v10.0.0-alpha.1 (/home/vyzo/src/fvm/builtin-actors/actors/evm)
error[E0502]: cannot borrow `*rt` as immutable because it is also borrowed as mutable
   --> actors/evm/src/lib.rs:165:55
    |
160 |         let mut system = System::load(rt, readonly).map_err(|e| {
    |                                       -- mutable borrow occurs here
...
165 |             crate::interpreter::system::load_bytecode(rt.store(), &cid)?
    |                                                       ^^^^^^^^^^ immutable borrow occurs here
...
174 |         let caller_fil_addr = system.rt.message().caller();
    |                               ------------------- mutable borrow later used here
```

that language....

Reply:
I am going to leave the argument to avoid the situation with the borrow, but introduce a helper to dedup the code.

Reply:
https://github.com/filecoin-project/builtin-actors/pull/768/commits/49300e1ed897aba442a13c78fd4917ca9e90b284